File: #03/Categoria 04/q13 - sublist/sublist.hs
sublist :: Int -> Int -> [a] -> [a]
sublist a b xs = drop x (take y xs)
  where
    size = length xs
    x = if a < 0 then a + size else a
    y = if b < 0 then b + size else b

main :: IO ()
main = do
    a <- readLn :: IO Int
    b <- readLn :: IO Int
    c <- readLn :: IO [Int]
    print $ sublist a b c

File: #03/Categoria 04/q13 - sublist/sublist.js
sublist = (a, b, arr) => arr.slice(a,b);


console.log(sublist(1,3,[0,1,2,3,4,5,6,7,8,9,10]));
console.log(sublist(0,11,[0,1,2,3,4,5,6,7,8,9,10]));
console.log(sublist(2,8,[0,1,2,3,4,5,6,7,8,9,10]));
console.log(sublist(0, -1,[0,1,2,3,4,5,6,7,8,9,10]));
console.log(sublist(2,-2,[0,1,2,3,4,5,6,7,8,9,10]));
console.log(sublist(-10,-1,[0,1,2,3,4,5,6,7,8,9,10]));
console.log(sublist(-4,-2,[0,1,2,3,4,5]));